src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,25 @@
+package duke.task;
+
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
+
+    public boolean getIsDone() {
+        return this.isDone;
+    }
+
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    public abstract String toString();
```

Comment:
You can define toString() here as returning the main few things in a 'task printout' - the checkbox and description. Then, you can use super.toString() in the subclasses of Task like so: `return "[D] " + super.toString() + " by: " + this.by`